### PR TITLE
Cleaning SmartStereoProjectionPoseFactor

### PR DESCRIPTION
**Pull request by ToniRV**
- ToDo:
- [x] Split .h/.cpp
- [x] Use const&, avoid needless and costly copies
- [x] Lint everything to 80 char otw it's annoying to read.
- [x] Add asserts where the user could mess up badly (i.e. different sizes for measurements and keys, etc)
- [x] had to add gtsam_unstable to examples... Can I avoid that? Yes, moved the example to gtsam_unstable
- [x] calibration() returns a weird const vector, if const of the K_ elements was the intended purpose, the const should be inside the shared_ptr... otherwise the original const is useless.
- [ ] Add a unique Id (Key) to the smart factor, otw there is a ton of bookkeeping to be done by the user (perhaps in another PR)

### CODE REVIEW

**Comment by ProfFan** [comment]
@ToniRV There is a dedicated examples for unstable: 
https://github.com/borglab/gtsam/tree/develop/gtsam_unstable/examples

**Comment by ToniRV** [comment]
@ProfFan I guess the problem then is that [examples/ISAM2_SmartFactorStereo_IMU.cpp](https://github.com/borglab/gtsam/blob/develop/examples/ISAM2_SmartFactorStereo_IMU.cpp)
should be in the `gtsam_unstable/examples` folder instead of `examples`?

**Comment by dellaert** [comment]
I think that’s right. Feel free to move it!
thanks for this PR!

File: gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /* ----------------------------------------------------------------------------
 
- * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * GTSAM Copyright 2010, Georgia Tech Research Corpo
+ * ation,
```
Comment (dellaert) [inline gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h:4]: Fix

File: gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h [hunk 2]
```diff
@@ -28,8 +30,10 @@ namespace gtsam {
  * @addtogroup SLAM
  *
  * If you are using the factor, please cite:
- * L. Carlone, Z. Kira, C. Beall, V. Indelman, F. Dellaert, Eliminating conditionally
- * independent sets in factor graphs: a unifying perspective based on smart factors,
+ * L. Carlone, Z. Kira, C. Beall, V. Indelman, F. Dellaert, Eliminating
```
Comment (dellaert) [inline gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h:33]: Nicely format after auto format?

**Review by dellaert — APPROVED** [review]
I’d move the example to unstable, but other than that, looks great!

File: gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h (hunk 2 above)
Comment (ToniRV) [reply gtsam_unstable/slam/SmartStereoProjectionPoseFactor.h:33]: Yes, `clang` 80char limit. Fixed now.

**Review by dellaert — APPROVED** [review]
Cool. I'll merge.

**Pull request merged**